### PR TITLE
Implement movie sections 6

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
+import { createMovieSalesAgentDeal, MovieSalesAgentDeal } from '../../+state';
+import { getLabelByCode } from '../../staticModels';
+
+@Component({
+  selector: '[agentDeal] movie-display-sales-agentdeal',
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.ts]: Maybe `sales-display-deal` ?

File: libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  templateUrl: './sales-agent-deal.component.html',
+  styleUrls: ['./sales-agent-deal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MovieDisplaySalesAgentDealComponent {
+
+  public data : MovieSalesAgentDeal;
+  public getLabelByCode = getLabelByCode;
+
+  @Input() set agentDeal(agentDeal: Partial<MovieSalesAgentDeal>) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.ts]: maybe deal ?

File: libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <input matInput formControlName="rightsEnd" [matDatepicker]="picker" placeholder="Choose a date">
+    <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+    <mat-datepicker #picker></mat-datepicker>
+  </mat-form-field>
+  <!-- TERRITORIES -->
+  <h3> Mandate Territories </h3>
+  <mat-form-field class="full-width">
+    <mat-select formControlName="territories" placeholder="Territories" multiple>
+      <mat-option>
+        <ngx-mat-select-search [formControl]="territoriesFilterCtrl" [placeholderLabel]="'Search...'"
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html]: Do you really need this one ? Could you use angular material standard components instead. Or our own auto-chips-complete ?

File: libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        {{ territory.label }}
+      </mat-option>
+    </mat-select>
+  </mat-form-field>
+  <!-- MEDIAS -->
+  <h3> Mandate Medias </h3>
+  <mat-form-field class="full-width">
+    <mat-select formControlName="medias" placeholder="Medias" multiple>
+      <mat-option>
+        <ngx-mat-select-search [formControl]="mediasFilterCtrl" [placeholderLabel]="'Search...'"
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html]: Same here

File: libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+@Component({
+  selector: '[formGroup] movie-form-sales-agentdeal, [formGroupName] movie-form-sales-agentdeal',
+  templateUrl: './sales-agent-deal.component.html',
+  styleUrls: ['./sales-agent-deal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MovieFormSalesAgentDealComponent implements OnInit {
+  
+  public staticModels: any;
+  public territoriesFilterCtrl: FormControl = new FormControl();
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.ts]: Don't type it, if you do `new FormControl` just after

File: libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<ng-container *ngIf="data">
+  <mat-list>
+    <mat-list-item *ngIf="data.rightsEnd">
+      <strong>Mandate End of rights :</strong> {{data.rightsEnd | date: 'dd MMM yyyy' }}
+    </mat-list-item>
+    <mat-list-item *ngIf="data.territories.length > 0">
+      <strong>Territories :</strong>
+      <ul>
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/display/sales-agent-deal/sales-agent-deal.component.html]: This is temporary, did not do any html or css yes

File: libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html (hunk 3 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html]: This is temporary, did not do any html or css yet

File: libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html (hunk 4 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/form/sales-agent-deal/sales-agent-deal.component.html]: This is temporary, did not do any html or css yet

File: libs/movie/src/lib/movie/form/root/root.component.html [hunk 7]
```diff
@@ -27,6 +27,9 @@
         <mat-tab label="Festival Prizes">
           <movie-form-festivalprizes formGroupName="festivalPrizes"></movie-form-festivalprizes>
         </mat-tab>
+        <mat-tab label="Sales Agent Deal">
+          <movie-form-sales-agentdeal formGroupName="salesAgentDeal"></movie-form-sales-agentdeal>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/root/root.component.html]: Maybe sales-form-deal ? ^^

**Pull request merged**